packages/ag-charts-community/src/widget/widgetListenerInternal.ts

```diff
@@ -94,52 +94,39 @@ export class WidgetListenerInternal {
     destroy(): void {
         this.dragTriggerRemover?.();
         this.dragTriggerRemover = undefined;
-        this.dragStartListeners = undefined;
-        this.dragMoveListeners = undefined;
-        this.dragEndListeners = undefined;
+        this.listeners?.clear();
         this.mouseDragger?.destroy();
         this.touchDragger?.destroy();
     }
 
+    private getListenerSet<T extends Targetable, K extends EventType>(type: K): Set<EventHandler<T, K>> {
+        this.listeners ??= new Map();
+        let result: Set<EventHandler<T, K>> | undefined = this.listeners.get(type);
+        if (result === undefined) {
+            result = new Set();
+            this.listeners.set(type, result);
+        }
+        return result;
+    }
+
     add<T extends Targetable, K extends EventType>(type: K, target: T, handler: EventHandler<T, K>): void;
     add<T extends Targetable, K extends EventType>(type: K, target: T, handler: EventHandler<unknown>): void {
+        this.getListenerSet(type).add(handler);
         switch (type) {
-            case 'drag-start': {
-                this.dragStartListeners ??= [];
-                this.dragStartListeners.push(handler);
-                this.registerDragTrigger(target);
-                break;
-            }
-            case 'drag-move': {
-                this.dragMoveListeners ??= [];
-                this.dragMoveListeners.push(handler);
-                this.registerDragTrigger(target);
-                break;
-            }
+            case 'drag-start':
+            case 'drag-move':
             case 'drag-end': {
-                this.dragEndListeners ??= [];
-                this.dragEndListeners.push(handler);
                 this.registerDragTrigger(target);
                 break;
             }
+            default:
+                (type) satisfies 'close-widget';
```

Comment:
Might be worth looking into https://typescript-eslint.io/rules/switch-exhaustiveness-check/

Reply:
This might be more readable anyway

```suggestion
            case 'close-widget':
                break;
            default:
                (type) satisfies never;
```

Reply:
Well spotted! I've got a separate PR for this https://github.com/ag-grid/ag-charts/pull/3982, so I'll just remove this check for now.